mne_bids/tsv_handler.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from collections import OrderedDict, Counter
+
+
+def from_tsv(fname, dtypes=None):
+    """Read a tsv file into an OrderedDict
+
+    Parameters
+    ----------
+    fname : str
+        Path to the file being loaded.
+    dtypes : list
+        List of types to case the values loaded as. This is specified column by
+        column.
+        Defaults to None. In this case all the data is loaded as strings."""
+    data = np.loadtxt(fname, dtype=str, delimiter='\t')
+    column_names = data[0, :]
+    info = data[1:, :]
+    data_dict = OrderedDict()
+    if isinstance(dtypes, (list, tuple)):
+        if len(dtypes) == info.shape[1]:
+            for i, name in enumerate(column_names):
+                data_dict[name] = info[:, i].astype(dtypes[i]).tolist()
+    else:
+        for i, name in enumerate(column_names):
+            data_dict[name] = info[:, i].tolist()
+    return data_dict
+
+
+def to_tsv(data, fname):
+    """ Write an OrderedDict into a tsv file """
+    rows = len(data[list(data.keys())[0]])
+    with open(fname, 'w') as f:
+        f.write('\t'.join(list(data.keys())))
+        f.write('\n')
+        for i in range(rows):
+            row_data = list(str(data[key][i]) for key in data)
+            f.write('\t'.join(row_data))
+            f.write('\n')
+
+
+def combine(data1, data2, drop_column=None):
+    """Add two OrderedDict's together with the option of dropping repeated data
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Original array of data.
+    other : numpy.ndarray
+        New array of data.
+    drop_column : str
+        Name of the column to check for duplicate values in.
+        Any duplicates found will be dropped from the original data array (ie.
+        most recent value are kept).
+
+    """
+    for key, value in data2.items():
+        data1[key].extend(value)
+    if drop_column in data1:
+        column_data = data1[drop_column]
+        values_count = Counter(column_data)
+        for key, value in values_count.items():
+            # find the locations of the first n-1 values and remove the rows
+            for _ in range(value - 1):
+                idx = data1[drop_column].index(key)
+                for column in data1.values():
+                    del column[idx]
+
+
+def drop(data, values, column):
```

Comment:
I would make these all private functions. We don't want users accessing them accidentally. It's only for our purposes and we should have the freedom to refactor it later if needed